src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt

```diff
@@ -0,0 +1,50 @@
+package com.vanniktech.maven.publish
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.Upload
+
+internal abstract class BaseMavenPublishPlugin : Plugin<Project> {
+
+  override fun apply(p: Project) {
+    val extension = p.extensions.create("mavenPublish", MavenPublishPluginExtension::class.java, p)
+
+    val pom = MavenPublishPom.fromProject(p)
+    p.group = requireNotNull(pom.groupId) { "groupId is required to be set" }
+    p.version = requireNotNull(pom.version) { "version is required to be set" }
```

Comment:
Those two weren't necessarily required before, but setting null here and in the pom doesn't really make sense.